Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -1042,11 +1042,24 @@ ThrowCompletionOr<Value> left_shift(GlobalObject& global_object, Value lhs, Valu
         return Value(lhs_i32 << rhs_u32);
     }
     if (both_bigint(lhs_numeric, rhs_numeric)) {
+        // 6.1.6.2.9 BigInt::leftShift ( x, y ), https://tc39.es/ecma262/#sec-numeric-types-bigint-leftShift
         auto multiplier_divisor = Crypto::SignedBigInteger { Crypto::NumberTheory::Power(Crypto::UnsignedBigInteger(2), rhs_numeric.as_bigint().big_integer().unsigned_value()) };
-        if (rhs_numeric.as_bigint().big_integer().is_negative())
-            return Value(js_bigint(vm, lhs_numeric.as_bigint().big_integer().divided_by(multiplier_divisor).quotient));
-        else
-            return Value(js_bigint(vm, lhs_numeric.as_bigint().big_integer().multiplied_by(multiplier_divisor)));
+
+        // 1. If y < 0ℤ, then
+        if (rhs_numeric.as_bigint().big_integer().is_negative()) {
+            // a. Return the BigInt value that represents ℝ(x) / 2^-y, rounding down to the nearest integer, including for negative numbers.
```

Comment:
Also this was what I was asking in the discord just copying from the spec gives:
`Return the BigInt value that represents ℝ(x) / 2-y, rounding down to the nearest integer, including for negative numbers.`
Note in particular the `2-y` which is a little confusing and here more with the `-` sign

Reply:
`2^-y` was decided.